pkg/connector/roleMapping.go

```diff
@@ -0,0 +1,89 @@
+package connector
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/conductorone/baton-elastic/pkg/elastic"
+	v2 "github.com/conductorone/baton-sdk/pb/c1/connector/v2"
+	"github.com/conductorone/baton-sdk/pkg/annotations"
+	"github.com/conductorone/baton-sdk/pkg/pagination"
+	rs "github.com/conductorone/baton-sdk/pkg/types/resource"
+)
+
+type roleMappingBuilder struct {
+	resourceType         *v2.ResourceType
+	client               *elastic.Client
+	shouldSyncDeployment bool
+}
+
+func (d *roleMappingBuilder) ResourceType(ctx context.Context) *v2.ResourceType {
+	return d.resourceType
+}
+
+// Create a new connector resource for Elastic deployment user.
+func roleMappingResource(role string) (*v2.Resource, error) {
+	profile := map[string]interface{}{
+		"role_mapping_id":   role,
+		"role_mapping_name": role,
+	}
+
+	status := v2.UserTrait_Status_STATUS_UNSPECIFIED
+	userTraitOptions := []rs.UserTraitOption{
+		rs.WithUserProfile(profile),
+		rs.WithStatus(status),
+	}
+
+	ret, err := rs.NewUserResource(
```

Comment:
Are role mappings users? I thought they were some other kind of resource.

Reply:
Oh, it looks like role mappings are returned as a list of users. In that case they should be part of the role resource's grants function.

Reply:
Fixed